src/main/resources/db/migration/V1_132__print_files_to_clob_update.sql

```diff
@@ -0,0 +1,216 @@
+				v_documents_to_write := v_total_documents_size;
+			else
+				v_total_documents_size := v_total_documents_size - p_document_limit;
+				v_documents_to_write := p_document_limit;
+			end if;
+
+			-- create new header for new file
+			SELECT RPAD('   ' ||RPAD(p_form_type,16)||LPAD(v_documents_to_write::VARCHAR(20),6,'0')||LPAD(v_page_size::VARCHAR(20),6,'0')||'50'||LPAD(p_max_rec_len::VARCHAR(20),8,'0'),256,' ')
+			INTO v_header;
+
+			-- erasing data for new file
+			v_data := '' || v_header || chr(10);
+			p_count := 0;
+
+		END IF;
+
+		-- Form the data column details first
+		SELECT 	form_details.id,
+				form_details.detail_rec
+		INTO v_row_id, v_detail_rec;
+
+
+		-- If data is empty then the first line will be the header plus newline otherwise it will be the current data plus header plus newline
+		v_data := v_data||v_detail_rec||CHR(10);
+
+		UPDATE juror_mod.bulk_print_data
+		SET  extracted_flag = 'Y'
+		WHERE id = v_row_id;
+
+		-- increment the loop counter so that the process can be stopped if it reaches the threshold
+		p_count := p_count + 1;
+
+	END LOOP;
+
+	-- flush data to content store if the limit hasn't been reached
+	IF v_data <> '' THEN
+		INSERT INTO juror_mod.content_store(request_id,document_id,file_type,data)
+				VALUES (NEXTVAL('juror_mod.content_store_seq'),v_filename,'PRINT',v_data);
+				COMMIT;
+	END IF;
+END;
+$procedure$
+;
+
+--DROP PROCEDURE juror_mod.delete_printfiles();
+
+CREATE OR REPLACE PROCEDURE juror_mod.delete_printfiles()
+ LANGUAGE plpgsql
+AS $procedure$
+BEGIN
+	DELETE
+	FROM juror_mod.bulk_print_data bpd
+	WHERE COALESCE (bpd.extracted_flag,'N') = 'N'
+	AND EXISTS 	(
+					SELECT 1
+					FROM juror_mod.juror_pool jp
+					WHERE jp.juror_number = bpd.juror_no
+					AND jp.is_active = true
+					AND (
+							(bpd.form_type IN ('5224','5224C') AND jp.status <> 6) -- withdrawal letters
+			                OR
+			                (bpd.form_type IN ('5225','5225C') AND jp.status <> 5) -- excusal letters
+			                OR
+			                (bpd.form_type IN ('5226','5226C') AND jp.status <> 2) -- excusal denied letters
+			                OR
+			                (bpd.form_type IN ('5226A','5226AC') AND jp.status <> 2) -- deferral denied letters
+			                OR
+			                (bpd.form_type IN ('5227','5227C') AND jp.status <> 1) -- request for info letters
+			                OR
+			                (bpd.form_type IN ('5229','5229C') AND jp.status <> 7) -- postpone letters
+			                OR
+			                (bpd.form_type IN ('5229A','5229AC') AND jp.status not in (2,7)) -- deferral letters
+			             	OR
+			             	(bpd.form_type IN ('5229A','5229AC') AND jp.status = 2 AND COALESCE(jp.was_deferred,false) = false) -- deferral letters, deferral deleted
+						)
+				);
+END;
+$procedure$
+;
+
+-- return total pages for form type
+create or replace procedure juror_mod.get_form_type_page_length(in form_type varchar(6), inout page_size integer)
```

Comment:
Can we discuss this please. Page size is not something that I am familiar with so am not sure why we have this here.

Reply:
Chris to check what actually Heritage does.

Reply:
Heritage puts the number of records in the page size field as well in the number of records field, as shown below

 rpad('   ' || rpad(form_type,16) || lpad(record_count,6,0) || lpad(record_count),6,0) || '50' || lpad(recordlength,8,0),256,' ')

We should do the same as Heritage.

Reply:
Updated to use the record count for both